src/server/lib/breachedPasswordCheck.ts

```diff
@@ -0,0 +1,40 @@
+import { createHash } from 'crypto';
+import { fetch } from '@/server/lib/fetch';
+import { logger } from '@/server/lib/serverSideLogger';
+import { trackMetric } from '@/server/lib/trackMetric';
+
+export const isBreachedPassword = async (
+  password: string,
+): Promise<boolean> => {
+  try {
+    const hash = createHash('sha1').update(password).digest('hex');
+    const first5 = hash.substring(0, 5);
```

Comment:
a doc comment for this method / explanatory comment about the method of checking whether a password is breached could really help to illustrate the logic behind how the breached check works here

Reply:
Good idea! I'll add some additional comments.